app/controllers/sub_space_booking_controller.rb

```diff
@@ -84,16 +149,65 @@ def bookings
           end
         end
     end
-
     render json: @bookings
   end
 
   def create
+    unless signed_in?
+      redirect_to login_path, alert: "You must be logged in to view this page."
+      return
+    end
+    unless current_user.booking_approval || current_user.admin?
+      redirect_to root_path, alert: "You must be approved to book spaces."
+      return
+    end
+
+    if params[:sub_space_booking][:recurring].present?
```

Comment:
Idea: Maybe put that in a before_save or before_create in the model, more processing should be done on the model side, it would unclutter a bit the controller

Reply:
Edit: I am wrong